controllers/upgrade.js

```diff
@@ -0,0 +1,45 @@
+const PouchDB = require('pouchdb-core');
+PouchDB.plugin(require('pouchdb-adapter-http'));
+
+const buildDbUrl = 'https://staging.dev.medicmobile.org/_couch/builds';
+const buildDb = new PouchDB(buildDbUrl);
+const targetDbUrl = process.env.COUCH_URL;
+const targetDb = new PouchDB(targetDbUrl);
+
+module.exports = (version, username) => {
+  console.log('upgrade()', `Upgrading to ${version}…`);
+
+  console.log('upgrade()', 'Fetching newDdoc…');
+  return buildDb
+    .get(version, { attachments:true })
+    .then(newDdoc => console.log('upgrade()', 'Fetched newDdoc.') || newDdoc)
+
+    .then(newDdoc => console.log('upgrade()', 'Fetching oldDdoc…') || newDdoc)
+    .then(newDdoc =>
+      targetDb.get('_design/medic')
+        .then(oldDdoc => console.log('upgrade()', 'Fetched oldDdoc.') || oldDdoc)
+
+        .then(oldDdoc => {
+          newDdoc.app_settings = oldDdoc.app_settings;
```

Comment:
I think it's close, but if we ship new settings with defaults they'll be lost on upgrade. We need some sort of deep `_.defaults` so if we add a new nested configuration it'll merge in seamlessly. Something like (untested)...

```js
const deepDefaults = (oldObj, newObj) => {
  Object.keys(newObj).forEach(key => {
    if (typeof oldObj[key] === 'undefined') {
      return;
    }
    if (typeof oldObj[key] === 'object') {
      deepDefaults(oldObj[key], newObj[key]);
    } else {
      newObj[key] = oldObj[key];
    }
  });
  return newObj;
}
```